app/views/devise/shared/_links.erb

```diff
@@ -4,8 +4,12 @@
       <%- resource_class.omniauth_providers.each do |provider| %>
         <% if provider.to_s == "google_oauth2" && (current_page?(new_user_session_path) || current_page?(new_user_password_path)) %>
```

Comment:
TODO:
add a helper that handles showing of button links

show_signup_page? provider

show_register_page? provider